content/en/blog/_posts/2024-01-26-webassembly-on-kubernetes-from-containers-to-wasm/index.md

```diff
@@ -0,0 +1,73 @@
+
+## What is WebAssembly?
+
+WebAssembly is a universal bytecode technology that allows programs written in various languages like Go, Rust, and C/C++ to be compiled into bytecode, which can be executed directly within web browsers and servers.
+
+{{< figure src="01-webassembly-runs-on-browser-and-server.svg" alt="The diagram illustrates how programs written in languages like Go, Rust and C/C++ can be compiled to WebAssembly bytecode to run efficiently across browsers and servers." caption="Figure 1: WebAssembly runs on browser and server" >}}
+
+WebAssembly is designed from the ground up to solve the performance problem of JavaScript. With WebAssembly, developers can compile code to a low-level binary format that can be executed by modern web browsers at near-native speeds.
+
+In March 2019, Mozilla announced the _WebAssembly System Interface (_WASI_),_ an API specification that defines a standard interface between WebAssembly modules and their host environments. WASI allows Wasm modules to access system resources securely, including the network, filesystem, etc. This extremely expanded Webassembly's potential by enabling it to work not only in browsers but also on servers.
+
+## The advantages of WebAssembly
+
+WebAssembly stands out with several remarkable benefits over traditional containers:
+- **Fast**: Wasm modules typically start within milliseconds, significantly faster than traditional containers, which is crucial for workloads requiring rapid startup, such as serverless functions.
+- **Lightweight**: Compared to container images, Wasm modules generally occupy less space and demand fewer CPU and memory resources.
+- **Secure**: Wasm modules run in a strict sandbox environment, isolated from the underlying host operating system, reducing potential security vulnerabilities.
+- **Portable**: Wasm modules can run seamlessly across various platforms and CPU architectures, eliminating the need to maintain multiple container images tailored for different OS and CPU combinations.
+
+You can refer to this table for a detailed comparison between WebAssembly and containers: [WebAssembly vs Linux Container](https://wasmedge.org/wasm_linux_container/)。
+
+## Run Wasm modules in Linux containers
+
+An easy way to run Wasm modules in the container ecosystems is to embed the Wasm bytecode file in the Linux container image. Precisely, you can trim down the Linux OS inside the container to the point where it is just enough to support the Wasm runtime. Since Wasm modules are housed in standard containers, they can be integrated seamlessly with any existing container ecosystems.
+
+Compared to a regular Linux OS, the attack surface of the slimmed Linux OS is dramatically reduced. Nonetheless, this approach still necessitates the launching of a Linux container. Although the Linux OS is trimmed down, it still takes up 80% of the container's image size.
```

Comment:
I'm worried about plagiarism. The text here seems close to wording in https://wasmedge.org/docs/develop/deploy/using-wasmedge-in-docker/

Can you explain the similarity?

Reply:
Here, I refer to the documentation of WasmEdge.
I rephrase the sentences: https://github.com/kubernetes/website/pull/43760/commits/f9582e75d9a7bf208c4c6ba408906a3fbcd57223